altos-rust/altos-core/src/task/control.rs

```diff
@@ -63,6 +69,13 @@ impl Priority {
     (0..NUM_PRIORITIES)
   }
 
+
+  /// Returns a box of the range of values of Priority minus the one passed in.
+  pub fn all_except<'a>(priority:Priority) -> Box<Iterator<Item=usize> + 'a> {
+    let all = Priority::all();
+    Box::new(all.filter(move |&x| x != priority as usize))
```

Comment:
Vectors are also allocated on the heap so they'll cause the same problem as using Box.